src/preact/base-element.js

```diff
@@ -837,129 +933,30 @@ export class PreactBaseElement extends BaseElement {
- * @param {!Object} api
- * @param {string} key
- */
-function wrapRefProperty(baseElement, api, key) {
-  Object.defineProperty(api, key, {
-    configurable: true,
-
-    get() {
-      return baseElement.currentRef_[key];
-    },
-
-    set(v) {
-      baseElement.currentRef_[key] = v;
-    },
-  });
-}
 
-// Ideally, these would be Static Class Fields. But Closure can't even.
-
-/**
- * Override to provide the Component definition.
- *
- * @protected {!PreactDef.FunctionalComponent}
- */
-PreactBaseElement['Component'] = function () {
-  devAssert(false, 'Must provide Component');
-};
-
-/**
- * If default props are static, this can be used instead of init().
- * @protected {!JsonObject|undefined}
- */
-PreactBaseElement['staticProps'] = undefined;
-
-/**
- * @protected {!Array<!ContextProp>}
- */
-PreactBaseElement['useContexts'] = mode.isLocalDev() ? Object.freeze([]) : [];
-
-/**
- * Whether the component implements a loading protocol.
- *
- * @protected {boolean}
- */
-PreactBaseElement['loadable'] = false;
-
-/**
- * Whether a component should be unloaded for `pauseCallback`.
- *
- * @protected {boolean}
- */
-PreactBaseElement['unloadOnPause'] = false;
-
-/**
- * An override to specify that the component requires `layoutSizeDefined`.
- * This typically means that the element's `isLayoutSupported()` is
- * implemented via `isLayoutSizeDefined()`, and this is how the default
- * `isLayoutSupported()` is implemented when this flag is set.
- *
- * @protected {string}
- */
-PreactBaseElement['layoutSizeDefined'] = false;
-
-/**
- * The tag name, e.g. "div", "span", time" that should be used as a replacement
- * node for Preact rendering. This is the node that Preact will diff with
- * with specified, instead of rendering a new node. Only applicable to light-DOM
- * mapping styles.
- *
- * @protected {string}
- */
-PreactBaseElement['lightDomTag'] = '';
-
-/**
- * Whether this element uses "templates" system.
- *
- * @protected {boolean}
- */
-PreactBaseElement['usesTemplate'] = false;
-
-/**
- * The CSS for shadow stylesheets.
- *
- * @protected {?string}
- */
-PreactBaseElement['shadowCss'] = null;
-
-/**
- * Whether this element uses Shadow DOM.
- *
- * @protected {boolean}
- */
-PreactBaseElement['usesShadowDom'] = false;
-
-/**
- * Enabling detached mode alters the children to be rendered in an
- * unappended container. By default the children will be attached to the DOM.
- *
- * @protected {boolean}
- */
-PreactBaseElement['detached'] = false;
-
-/**
- * This enables the 'delegatesFocus' option when creating the shadow DOM for
- * this component.  A key feature of 'delegatesFocus' set to true is that
- * when elements within the shadow DOM gain focus, the focus is also applied
- * to the host element.
- */
-PreactBaseElement['delegatesFocus'] = false;
+  /**
+   * @param {Object} api
+   * @param {string} key
+   */
+  wrapRefProperty(api, key) {
+    // eslint-disable-next-line local/restrict-this-access
```

Comment:
Likely unneeded now that it's an instance method.

Reply:
`this` access is banned. it wasn't necessary when calling `this.currentRef_` _but_ that doesn't work b/c of the `this` changes within scope of the defineProperty getter/setter.

Reply:
Ah, fixed with arrow func. Removed now.

Reply:
Use arrows instead of methods